src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -0,0 +1,34 @@
+package com.squareup.javapoet;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public final class MethodSpecTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
```

Comment:
We don't use JUnit expected exceptions because they're bad. They don't constrain the line that the exception is thrown from, and they're magic framework that adds little value. See the try/catch clauses in the TypeSpecTest for our pattern.